Web of Science.js

```diff
@@ -0,0 +1,211 @@
+			}
+			if(creator.lastName.toUpperCase() == creator.lastName) {
+				creator.lastName = ZU.capitalizeTitle(creator.lastName, true);
+			}
+		}
+		item.complete();
+	});
+	importer.translate();
+}
+
+function fetchIds(ids, doc) {
+	var outputForm = getOutputForm(doc);
+	var postData = getHiddenValues(outputForm);
+	var filters = 'USAGEIND RESEARCHERID ACCESSION_NUM FUNDING SUBJECT_CATEGORY '
+		+ 'JCR_CATEGORY LANG IDS PAGEC SABBR CITREFC ISSN PUBINFO KEYWORDS '
+		+ 'CITTIMES ADDRS CONFERENCE_SPONSORS DOCTYPE ABSTRACT CONFERENCE_INFO '
+		+ 'SOURCE TITLE AUTHORS '
+		//additional fields from INSPEC
+		+ 'ADDRESS AUTHORS_EDITORS AUTHORSIDENTIFIERS CLASSIFICATION_CODES '
+		+ 'CONFERENCE_SPONSORS DESCRIPTORS IDENTIFYING_CODES IMAGES '
+		+ 'INVENTORS_ASSIGNEES IPC NUM_OF_REF PATENT_INFO SPONSORS TRANSLATORS '
+		+ 'TREATMENT UNCONTROLLED_TERMS';
+	postData['value(record_select_type)'] = 'selrecords';
+	postData['markFrom'] = '';
+	postData['markTo'] = '';
+	postData['fields_selection'] = filters;
+	postData['filters'] = filters;
+	postData['save_options'] = 'othersoftware';
+	postData['format'] = 'saveToRef';
+
+	//add selected items
+	var selectedIds = ids.join(';');
+	postData['selectedIds'] = selectedIds;
+
+	var postUrl = outputForm.action;
+	Z.debug("Posting to " + postUrl);
+	ZU.doPost(postUrl, serializePostData(postData), function (text) {
+		//check if there's an intermediate page
+		if(text.indexOf('FN ') === 0) {
+			importISIRecord(text);
+			return;
+		}
+		
+		//otherwise we have an intermediate page (maybe... it just kind of went away one day)
+		//everything it mostly the same as above except for a few fields
+		var postData2 = {}
+		postData2['locale'] = postData['locale'];
+		postData2['colName'] = postData['colName'];
+		postData2['sortBy'] = postData['sortBy'];
+		postData2['SID'] = postData['SID'];
+		postData2['filters'] = postData['filters'];
+		postData2['fileOpt'] = 'fieldtagged';
+		postData2['action'] = 'saveDataToRef';
+		postData2['product'] = 'UA';
+		postData2['numRecords'] = ids.length;
+		postData2['numRecsToRetrieve'] = 500;
+		
+		var qid = text.match(/<input[^>]+name=(['"]?)qid\1[\s\/][^>]*/);
+		if(qid) qid = qid[0].match(/value=['"]?(\d+)/);
+		if(qid) {
+			qid = qid[1];
+		} else {
+			qid = postData['qid']*1+1;	//this can be wrong if pages are refreshed
+			Z.debug("Could not find qid on page. Using 1 + previous qid: " + qid);
+			text = text.replace(/\s*[\r\n]\s*/g, '\n');	//trim out the extra newlines
+			var forms = text.match(/<form[\s\S]+?<\/form>/ig);
+			if(forms) {
+				Z.debug("Page contained the following forms:");
+				Z.debug(forms.join('\n==============================\n'));
+			} else {
+				Z.debug("Could not find any forms on the page. Here's the whole HTML");
+				Z.debug(text);
+			}
+		}
+		postData2['qid'] = qid;
+
+		var postUrl2 = 'http://ets.webofknowledge.com/ETS/saveDataToRef.do';	//Zotero should take care of proxies
```

Comment:
This cross-origin request would be causing problems for IE, but it doesn't look like we hit this part any more. Interestingly, using the export form through the page results in the request being redirected to a cross-origin address, which would also be an issue for Safari, but it looks like somehow our way of POSTing for metadata avoids this redirect. Just something to keep in mind.